### DISCUSSION TITLE
When using multiple NetStream objects

### DISCUSSION THREAD

**Discussion by koreaflash** [airsdk/Adobe-Runtime-Support#1315] [Q&A]
When using multiple NetStream objects, only the first one plays normally, the next one throws a NetStream.Play.Failed error and playback cannot start.
This happens from the Air SDK version 32.0.0.168 to the latest version 33.1.1.633.

Is it normal that I can't use NetStream objects at the same time?

Air SDK version 32.0.0.89 and earlier versions It plays for a while and then(about 7 sec) stops. It cannot be played again after it is stopped.
No errors are issued and it just hangs. The state of the stream is "playing" . If you resume after pausing, the stream status will change to "playing" after "buffering", but it will not play.

Tested on an Android device.
There is no problem when using rtmp://.
Problems occur when using http:// and https://.
We play mp4 files.

**Comment by ajwfrost** [comment]
I suspect this is to do with the removal of the software video decoders from the AIR SDK when Adobe handed it over to us (and their final v32 build where they had done the preparatory work for this). Normally I believe you get one hardware-accelerated video stream available from the OS, and subsequent streams were then handled in software. With software being unavailable now, it would mean that these other ones would fail.

For the issues with earlier versions, I don't know for sure but suspect there may be some thread deadlock problems happening there.

I'm curious about the RTMP comment though: have you got multiple H.264 streams coming via RTMP and playing back with the later SDKs? or are these using a different codec? Or was that comment just referring to the hanging problems, rather than the fact that you can't play back a second NetStream stream?

thanks

**Reply by koreaflash** [reply]
I was calling mynetstream.dispose() when it became useless when using rtmp.
Even when streaming with http is implemented, dispose() is called and it works normally.
thank you.
